java/src/test/java/org/rocksdb/util/ByteBufferAllocator.java

```diff
@@ -0,0 +1,28 @@
+package org.rocksdb.util;
+
+import java.nio.ByteBuffer;
+
+public interface ByteBufferAllocator {
+  ByteBuffer allocate(int capacity);
+
+  ByteBufferAllocator DIRECT = new DirectByteBufferAllocator();
+  ByteBufferAllocator INDIRECT = new IndirectByteBufferAllocator();
```

Comment:
seems that "heap buffer" is more general

Reply:
@javeme If you mean in so far as naming, then we used direct/indirect because of Java's naming of these things: https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html#isDirect--

If you want us to move away from this naming, we can, but we need to agree on the names. As I understand it, both are technically allocated from the Heap, it's just that one is outside of the Java Heap.

Reply:
I agree both are allocated from the Heap, and not mean that indirect is incorrect.
I consider that the two commonly used subclasses of ByteBuffer are named HeapByteBuffer and DirectByteBuffer, so if the name HeapByteBufferAllocator is better understood for me. It's just a suggestion~

Reply:
I've changed the naming to Direct and Heap.